### PR TITLE
[TR] PIM-4797: Hide the Associations tab when I am not the owner

### PR DESCRIPTION
**Pull request by grena**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | Y
| Specs & Static     | Y
| Changelog updated  | N

The purpose of changes in `ProductEditDataFilter.php` is to have this class really easy to override.

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php [hunk 1]
```diff
@@ -159,13 +167,119 @@ protected function getNewValuesData(AttributeInterface $attribute, array $values
...
     /**
+     * Return whether the current user is allowed to update the given modification $type
+     * on the given $product
+     *
+     * @param string           $type
+     * @param ProductInterface $product
+     *
+     * @return bool
+     */
+    protected function isAllowed($type, ProductInterface $product)
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: Typehint first

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php [hunk 2]
```diff
@@ -159,13 +167,119 @@ protected function getNewValuesData(AttributeInterface $attribute, array $values
...
+    {
+        $isAllowed = false;
+
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: remove this line

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php [hunk 3]
```diff
@@ -159,13 +167,119 @@ protected function getNewValuesData(AttributeInterface $attribute, array $values
...
+     * @param $type
+     *
+     * @return bool
+     */
+    protected function checkAclForType($type)
+    {
+        $acl = $this->getAclForType($type);
+        $actionGranted = null === $acl || $this->securityFacade->isGranted($acl);
+
+        return $actionGranted;
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: return (null === $acl) || $this->securityFacade->isGranted($acl);

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php [hunk 4]
```diff
@@ -159,13 +167,119 @@ protected function getNewValuesData(AttributeInterface $attribute, array $values
...
+                break;
+            case 'associations':
+                $isAllowed = $this->isAllowedToUpdateAssociations($product);
+                break;
+            case 'values':
+                $isAllowed = $this->isAllowedToUpdateValues();
+                break;
+        }
+
+        return $isAllowed;
```
Comment (fitn) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: return $this->checkAclForType($type); ?

**Comment by solivier** [comment]
GTM

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php [hunk 5]
```diff
@@ -159,13 +167,118 @@ protected function getNewValuesData(AttributeInterface $attribute, array $values
...
+    }
+
+    /**
+     * Return whether the current user is allowed to update associations of the product
+     *
+     * @param ProductInterface $product
+     *
+     * @return bool
+     */
+    protected function isAllowedToUpdateAssociations(ProductInterface $product)
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: $product seems to be not used

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php [hunk 6]
```diff
@@ -159,13 +167,118 @@ protected function getNewValuesData(AttributeInterface $attribute, array $values
...
+     * Return whether the current user is allowed to update categories of the product
+     *
      * @param ProductInterface $product
      *
      * @return bool
      */
     protected function isAllowedToClassify(ProductInterface $product)
     {
-        return true;
+        return $this->checkAclForType('categories');
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: $product seems to be not used

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/associations.js [hunk 7]
```diff
@@ -359,6 +360,15 @@ define(
                 });
 
                 return assocTarget;
+            },
+
+            /**
+             * Set if this extension is visible
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/associations.js]: "Set" is not correct here, "Check" maybe ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/categories.js [hunk 8]
```diff
@@ -166,6 +167,15 @@ define(
                 }
 
                 return this.cache[id].code;
+            },
+
+            /**
+             * Set if this extension is visible
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/categories.js]: same

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php (hunk 6 above)
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: It's used for overriding I guess.
But I agree that from the CE point of view, passing the product in some methods and not in others seems strange.

**Comment by sumbobyboys** [comment]
GTM

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php (hunk 6 above)
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: Yes, used for overriding.
I take if someone has a better idea.

**Pull request merged**